framework/wazuh/core/indexer/__init__.py

```diff
@@ -0,0 +1,120 @@
+import os
+import random
+from asyncio import sleep
+from logging import getLogger
+
+from opensearchpy import AsyncOpenSearch
+from wazuh.core.exception import WazuhIndexerError
+
+logger = getLogger('wazuh')
+
+HOST_KEY = 'host'
+PORT_KEY = 'port'
+
+
+class Indexer:
+    """Interface to connect with Wazuh Indexer."""
+
+    def __init__(
+        self,
+        host: str,
+        user: str,
+        password: str,
+        port: int = 9200,
+        use_ssl: bool = True,
+        verify_certs: bool = True,
+        ca_certs: str = '',
```

Comment:
Is there a way of passing custom certificates (server key and certificate)? We should include those parameters if that's a thing

Reply:
Do you mean to authenticate? Instead of using HTTP basic auth?

Reply:
I was referring to the possibility of providing a custom certificate and key for the connection. Apparently the underlying HTTP client class accepts `client_key` and `client_cert` (I mentioned server credentials but in this case is client, my bad): https://github.com/opensearch-project/opensearch-py/blob/a029343fb58d752c06b6653d0718f73a1f7a182e/opensearchpy/connection/http_async.py#L33.

I think they should be included because if a user deploys the indexer using custom certificates (like the [documentation](https://documentation.wazuh.com/current/installation-guide/wazuh-indexer/step-by-step.html#certificates-creation) suggests), they won't be able to connect to it.